pkg/diagnostics/grpc_monitoring.go

```diff
@@ -227,6 +227,7 @@ func (g *grpcMetrics) UnaryServerInterceptor() func(ctx context.Context, req int
 			size = g.getPayloadSize(resp)
 		}
 		g.ServerRequestSent(ctx, info.FullMethod, status.Code(err).String(), int64(g.getPayloadSize(req)), int64(size), start)
+		RecordErrorCode(err)
```

Comment:
Let's check if err is nil before calling the record function

Reply:
wdyt on adding a nil check inside `RecordErrorCode`?

like:
```
func RecordErrorCode(err error) bool {
  if err == nil { return false}
  ...
}
```

Reply:
but no need to change more in any case

Reply:
Yep, even better!